app/components/Snackbar/index.js

```diff
@@ -112,64 +125,82 @@ class Snackbar extends React.Component {
 
   onDismissMessage() {
     const state = this.state;
-    this.setState({ ...state, message: null });
-    this.props.onDismissAllMessages();
-    this.clearHideTimer();
+    const messages = this.state.messages;
+    var newMessages = messages;
+    newMessages.pop();
+    this.setState({ ...state, messages: newMessages });
```

Comment:
Set progress to 0 here, so that we don't see a flash of a filled progress on the new notification

```suggestion
    this.setState({ ...state, progress: 0, messages: newMessages });
```